app/services/FileNotificationDetailsRetriever.scala

```diff
@@ -21,6 +21,6 @@ import model.{QuarantinedFile, S3ObjectLocation, UploadedFile}
 import scala.concurrent.Future
 
 trait FileNotificationDetailsRetriever {
-  def retrieveUploadedFileDetails(objectLocation: S3ObjectLocation): Future[UploadedFile]
-  def retrieveQuarantinedFileDetails(objectLocation: S3ObjectLocation): Future[QuarantinedFile]
+  def retrieveUploadedFileDetails(implicit objectLocation: S3ObjectLocation): Future[UploadedFile]
```

Comment:
Why implicit here?

Reply:
The implementation of that method needs an implicit S3ObjectLocation in order to call an implicit function - util.logging.LoggingDetails.fromS3ObjectLocation.